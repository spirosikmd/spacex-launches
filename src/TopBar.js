import React from 'react';
import { Link } from '@reach/router';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import GroupWorkRoundedIcon from '@material-ui/icons/GroupWorkRounded';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  logo: {
    marginLeft: -theme.spacing.unit,
    marginRight: theme.spacing.unit,
  },
  link: {
    textDecoration: 'none',
    color: theme.palette.common.white,
    display: 'flex',
    alignItems: 'center',
  },
});

const TopBar = ({ classes }) => {
  return (
    <AppBar position="static">
      <Toolbar>
        <Link to="/" className={classes.link}>
          <GroupWorkRoundedIcon className={classes.logo} />
          <Typography variant="h6" color="inherit">
            SpaceX Launches
          </Typography>
        </Link>
      </Toolbar>
    </AppBar>
  );
};

export default withStyles(styles)(React.memo(TopBar));
