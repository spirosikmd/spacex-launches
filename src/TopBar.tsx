import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import GroupWorkRoundedIcon from '@material-ui/icons/GroupWorkRounded';
import {
  withStyles,
  Theme,
  createStyles,
  WithStyles,
} from '@material-ui/core/styles';

const styles = (theme: Theme) =>
  createStyles({
    logo: {
      marginLeft: -theme.spacing.unit,
      marginRight: theme.spacing.unit,
    },
  });

const TopBar = ({ classes }: WithStyles<typeof styles>) => {
  return (
    <AppBar position="static">
      <Toolbar>
        <GroupWorkRoundedIcon className={classes.logo} />
        <Typography variant="h6" color="inherit">
          SpaceX Launches
        </Typography>
      </Toolbar>
    </AppBar>
  );
};

TopBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(React.memo(TopBar));
