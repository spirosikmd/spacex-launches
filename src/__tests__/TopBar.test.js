import TopBar from '../TopBar';

describe('TopBar', () => {
  let props;

  beforeEach(() => {
    props = {
      classes: {
        logo: 'logo',
      },
    };
  });

  it('renders', () => {
    expect(mountComponent(TopBar, props)).toMatchSnapshot();
  });
});
