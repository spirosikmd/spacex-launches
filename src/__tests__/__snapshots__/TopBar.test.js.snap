// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <ForwardRef
      to="/"
    >
      <pure(GroupWorkRoundedIcon)
        className="logo"
      />
      <WithStyles(Typography)
        color="inherit"
        variant="h6"
      >
        SpaceX Launches
      </WithStyles(Typography)>
    </ForwardRef>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
